src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,63 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final String WHITESPACE_CHARACTER = " ";
+    private static final String COMMA_CHARACTER = ",";
+    private static final String BUY_STRING = "buy";
+    private static final String SUPPLY_STRING = "supply";
+    private static final String RESULT_STRING = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] arrayOfData = readDataFromFile(fromFileName);
+        int[] dataForReport = makeReportFromFileData(arrayOfData);
+        writeDataToFile(toFileName, dataForReport);
+    }
+
+    private String[] readDataFromFile(String fromFileName) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            StringBuilder builder = new StringBuilder();
+            String value = reader.readLine();
+            while (value != null) {
+                builder.append(value).append(WHITESPACE_CHARACTER);
+                value = reader.readLine();
+            }
+            return builder.toString().split(WHITESPACE_CHARACTER);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read data from file: " + fromFileName, e);
+        }
+    }
+
+    private int[] makeReportFromFileData(String[] data) {
```

Comment:
let's return String (report) from this method
the method is supposed to make a report (as String)